controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -0,0 +1,493 @@
+		scrType         string
+		profileName     string
+		failureExpected bool
+		createObjects   []runtime.Object
+	}{
+		{
+			label:       "Reconcile blockpool StorageClassRequest",
+			scrType:     "blockpool",
+			profileName: fakeStorageProfile.Name,
+		},
+		{
+			label:       "Reconcile sharedfilesystem StorageClassRequest",
+			scrType:     "sharedfilesystem",
+			profileName: fakeStorageProfile.Name,
+		},
+		{
+			label:   "Reconcile blockpool StorageClassRequest with default StorageProfile",
+			scrType: "blockpool",
+		},
+		{
+			label:   "Reconcile sharedfilesystem StorageClassRequest with default StorageProfile",
+			scrType: "sharedfilesystem",
+		},
+		{
+			label:           "Reconcile blockpool StorageClassRequest with invalid StorageProfile",
+			scrType:         "blockpool",
+			profileName:     "nope",
+			failureExpected: true,
+		},
+		{
+			label:           "Reconcile sharedfilesystem StorageClassRequest with invalid StorageProfile",
+			scrType:         "sharedfilesystem",
+			profileName:     "nope",
+			failureExpected: true,
+		},
+	}
+
+	for _, c := range primaryTestCases {
+		caseCounter++
+		caseLabel := fmt.Sprintf("Case %d: %s", caseCounter, c.label)
+		fmt.Println(caseLabel)
+
+		r := createFakeReconciler(t)
+
+		fakeStorageClassRequest := &v1alpha1.StorageClassRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-scr",
+				Namespace: "test-ns",
+			},
+			Spec: v1alpha1.StorageClassRequestSpec{
+				Type:           c.scrType,
+				StorageProfile: c.profileName,
+			},
+		}
+		err = controllerutil.SetOwnerReference(fakeStorageConsumer, fakeStorageClassRequest, r.Scheme)
+		assert.NoError(t, err, caseLabel)
+
+		c.createObjects = append(c.createObjects, fakeCephFs)
+		c.createObjects = append(c.createObjects, fakeStorageClassRequest)
+		c.createObjects = append(c.createObjects, fakeStorageCluster)
+		c.createObjects = append(c.createObjects, fakeStorageConsumer)
+		c.createObjects = append(c.createObjects, fakeStorageProfile)
+
+		r.Cache = &informertest.FakeInformers{Scheme: r.Scheme}
+		fakeClient := fake.NewClientBuilder().WithScheme(r.Scheme).WithRuntimeObjects(c.createObjects...).WithStatusSubresource(fakeStorageClassRequest)
+		r.Client = fakeClient.Build()
+
+		req := reconcile.Request{}
+		req.Name = fakeStorageClassRequest.Name
+		req.Namespace = fakeStorageClassRequest.Namespace
+		_, err = r.Reconcile(context.TODO(), req)
+		if c.failureExpected {
+			assert.Error(t, err, caseLabel)
+			continue
+		}
+		assert.NoError(t, err, caseLabel)
+	}
+
+	caseCounter++
+	caseLabel := fmt.Sprintf("Case %d: No StorageClassRequest exists", caseCounter)
+	fmt.Println(caseLabel)
+
+	r := createFakeReconciler(t)
+	r.Cache = &informertest.FakeInformers{Scheme: r.Scheme}
+	r.Client = fake.NewClientBuilder().WithScheme(r.Scheme).Build()
+
+	req := reconcile.Request{}
+	req.Name = "nope"
+	req.Namespace = r.OperatorNamespace
+	_, err = r.Reconcile(context.TODO(), req)
+	assert.NoError(t, err, caseLabel)
```

Comment:
I might be wrong, doesn't existing watches stop these requests getting queued in the first place or are we testing something different, like reconciler just runs with no SCR in the store?